app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class AccountsController < ApplicationController
+  before_action :authorize_user
+
+  def index
+    render json: current_user.accounts, status: :ok
+  end
+
+  def show
+    return render_account_not_found if account.blank?
+
+    render json: account, status: :ok
+  end
+
+  def create
+    new_account = current_user.accounts.new(account_params)
+
+    if new_account.save
+      render json: { message: 'Account created with success' }, status: :created
+    else
+      render json: { errors: new_account.errors.full_messages },
+             status: :unprocessable_entity
+    end
+  end
+
+  def update
+    return render_account_not_found if account.blank?
+
+    if account.update(account_params)
+      render json: { message: 'Account updated with success' }, status: :ok
+    else
+      render json: { errors: @account.errors.full_messages },
```

Comment:
Aqui não seria sem o @?
```suggestion
      render json: { errors: account.errors.full_messages },
```

Reply:
Na verdade, no fim, não vai fazer diferença, mas... é melhor